src/main/java/com/flowingcode/vaadin/addons/demo/TabbedDemo.java

```diff
@@ -60,28 +60,25 @@ public TabbedDemo() {
     orientationCB = new Checkbox("Toggle Orientation");
     orientationCB.setValue(true);
     orientationCB.addClassName("smallcheckbox");
-    orientationCB.addValueChangeListener(
-        ev -> {
-          if (ev.getValue()) {
-            setOrientation(Orientation.HORIZONTAL);
-          } else {
-            setOrientation(Orientation.VERTICAL);
-          }
-        });
+    orientationCB.addValueChangeListener(ev -> {
+      if (ev.isFromClient()) {
+        toggleSplitterOrientation();
+      }
+    });
     codeCB = new Checkbox("Show Source Code");
     codeCB.setValue(true);
     codeCB.addClassName("smallcheckbox");
-    codeCB.addValueChangeListener(
-        ev -> {
-          setSourceVisible(ev.getValue());
-        });
+    codeCB.addValueChangeListener(ev -> {
+      if (ev.isFromClient()) {
```

Comment:
Shouldn't it call `updateSplitterPosition` regardless of whether the event is from client or not? (so that `setSourceVisible` updates the trigger position)

```diff
public void setSourceVisible(boolean visible) {
+  codeCB.setValue(visible);
-  sourceVisible = visible;
-  if (currentLayout != null) {
-    currentLayout.setSplitterPosition(visible ? 50 : 100);
-    orientationCB.setEnabled(visible);
-  }
}

+private void updateSplitterPosition() {
+  if (currentLayout != null) {
+    currentLayout.setSplitterPosition(codeCB.getValue() ? 50 : 100);
+    orientationCB.setEnabled(codeCB.getValue());
+  }
+}
```